File: appambit_sdk_flutter/android/src/main/kotlin/com/appambit/appambit_sdk_flutter/AnalyticsFlutter.kt
package com.appambit.appambit_sdk_flutter

import android.content.Context
import com.appambit.sdk.Analytics
import io.flutter.embedding.engine.plugins.FlutterPlugin
import io.flutter.plugin.common.MethodChannel

class AnalyticsFlutter {

    private lateinit var channel: MethodChannel
    private lateinit var context: Context

    fun attach(binding: FlutterPlugin.FlutterPluginBinding, context: Context) {
        this.context = context
        channel = MethodChannel(binding.binaryMessenger, "com.appambit/analytics")
        channel.setMethodCallHandler { call, result ->
            when (call.method) {
                "enableManualSession" -> {
                    Analytics.enableManualSession()
                    result.success(null)
                }
                "startSession" -> {
                    Analytics.startSession()
                    result.success(null)
                }
                "endSession" -> {
                    Analytics.endSession()
                    result.success(null)
                }
                "setUserId" -> {
                    val args = call.arguments as? Map<*, *>
                    val userId = args?.get("userId") as? String
                    Analytics.setUserId(userId)
                    result.success(null)
                }
                "setEmail" -> {
                    val args = call.arguments as? Map<*, *>
                    val email = (args?.get("email") as? String) ?: (args?.get("userEmail") as? String)
                    if (email != null) {
                        Analytics.setUserEmail(email)
                    }
                    result.success(null)
                }
                "generateTestEvent" -> {
                    Analytics.generateTestEvent()
                    result.success(null)
                }
                "clearToken" -> {
                    Analytics.clearToken()
                    result.success(null)
                }
                "trackEvent" -> {
                    val args = call.arguments as? Map<*, *>
                    val name = (args?.get("name") as? String) ?: (args?.get("eventTitle") as? String)
                    val props = asStringMap((args?.get("properties") as? Map<*, *>) ?: (args?.get("data") as? Map<*, *>))
                    if (name != null) {
                        if (props.isNotEmpty()) {
                            Analytics.trackEvent(name, props)
                        } else {
                            Analytics.trackEvent(name, null)
                        }
                        result.success(null)
                    } else {
                        result.notImplemented()
                    }
                }
                else -> result.notImplemented()
            }
        }
    }

    fun detach() {
        channel.setMethodCallHandler(null)
    }

    private fun asStringMap(map: Map<*, *>?): Map<String, String> {
        if (map == null) return emptyMap()
        val out = mutableMapOf<String, String>()
        for ((k, v) in map) {
            val key = k as? String ?: continue
            val value = v?.toString() ?: continue
            out[key] = value
        }
        return out
    }
}

File: appambit_sdk_flutter/ios/Classes/AnalyticsFlutter.swift
import Foundation
import Flutter
import AppAmbitSdk

final class AnalyticsFlutter {

  static func setUserId(args: Any?, result: @escaping FlutterResult) {
    guard
      let dict = args as? [String: Any],
      let userId = dict["userId"] as? String
    else {
      result(FlutterError(code: "BAD_ARGS", message: "Missing 'userId'", details: nil))
      return
    }
    Analytics.setUserId(userId) { err in
      if let err { result(FlutterError(code: "SET_USER_ID_ERR", message: err.localizedDescription, details: nil)) }
      else { result(nil) }
    }
  }

  static func setEmail(args: Any?, result: @escaping FlutterResult) {
    guard
      let dict = args as? [String: Any],
      let email = dict["email"] as? String
    else {
      result(FlutterError(code: "BAD_ARGS", message: "Missing 'email'", details: nil))
      return
    }
    // Tu implementación actual no llama completion; resolvemos inmediatamente
    Analytics.setEmail(email) { _ in }
    result(nil)
  }

  static func clearToken(result: @escaping FlutterResult) {
    Analytics.clearToken()
    result(nil)
  }

  static func startSession(result: @escaping FlutterResult) {
    Analytics.startSession { err in
      if let err { result(FlutterError(code: "START_SESSION_ERR", message: err.localizedDescription, details: nil)) }
      else { result(nil) }
    }
  }

  static func endSession(result: @escaping FlutterResult) {
    Analytics.endSession { err in
      if let err { result(FlutterError(code: "END_SESSION_ERR", message: err.localizedDescription, details: nil)) }
      else { result(nil) }
    }
  }

  static func enableManualSession(result: @escaping FlutterResult) {
    Analytics.enableManualSession()
    result(nil)
  }

  static func trackEvent(args: Any?, result: @escaping FlutterResult) {
    guard
      let dict = args as? [String: Any],
      let name = dict["name"] as? String,
      let props = dict["properties"] as? [String: String]
    else {
      result(FlutterError(code: "BAD_ARGS",
                          message: "Expected {name:String, properties:Map<String,String>}",
                          details: nil))
      return
    }


    Analytics.trackEvent(
      eventTitle: name,
      data: props
    ) { err in
      if let err { result(FlutterError(code: "TRACK_EVENT_ERR", message: err.localizedDescription, details: nil)) }
      else { result(nil) }
    }
  }

  static func generateTestEvent(result: @escaping FlutterResult) {
    Analytics.generateTestEvent { err in
      if let err { result(FlutterError(code: "GEN_TEST_EVENT_ERR", message: err.localizedDescription, details: nil)) }
      else { result(nil) }
    }
  }
}
